### PR TITLE
Consumer trait

### PR DESCRIPTION
**Pull request by sylfabre**
- fixes a bug in some implementation where the timeout is not respected (last wait should check for left time)
- implements a polling interval to fetch messages
- implements methods to set and get a visibility timeout

### CODE REVIEW

**Comment by makasim** [comment]
@sylfabre Could you please rebase, cuz I've merged message trait cherry pick to 0.7

File: src/Impl/ConsumerPollingTrait.php [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+declare(strict_types=1);
+
+namespace Interop\Queue\Impl;
+
+use Interop\Queue\Message;
+
+/**
+ * The ConsumerPollingTrait is a common implementation of the polling algorithm for a Consumer
+ */
+trait ConsumerPollingTrait
```
Comment (makasim) [inline src/Impl/ConsumerPollingTrait.php]: I would love to merge this trait and its tests. Could you make a separate PR with only this one?

**Comment by sylfabre** [comment]
@makasim I've updated the PR with the last 0.7 version

**Comment by makasim** [comment]
@PaulRouss3l you've just merged it, you could rely on it

**Comment by PaulRouss3l** [comment]
Thanks !
